[14.0][ADD] stock_mass_scrap

jcoux opened the pull request:
Add wizard to mass scrap expired products:

* Using the wizard "Mass scrap" you can create a scrap for each expired quant.
* Using the server action, you can mass validate all selected scraps

jcoux committed:
[14.0][ADD] stock_mass_scrap

Add wizard to mass scrap expired products:

* Using the wizard "Mass scrap" you can create a scrap for each expired quant.
* Using the server action, you can mass validate all selected scraps

omarie-ctc reviewed (approved):
After testing, fonctionnal part is ok.

AaronHForgeFlow commented:
LG. I'd like to have an option to exclude some of the lots that have expired, so the user can keep some of the lots.

rousseldenis commented on stock_mass_scrap/wizard/mass_scrap.py:
```diff
@@ -0,0 +1,91 @@
+            scrap_ids.append(scrap.id)
+
+        action = self.env.ref("stock.action_stock_scrap")
+        action_data = action.read()[0]
```
Please change to _for_xml_id()

jcoux replied:
I looked the function you indicate:
https://github.com/odoo/odoo/blob/8a02942f0cee4559db4466a8b656eb4e21c662fd/odoo/addons/base/models/ir_actions.py#L127
But I don't understand how you want I use it.

My need here is to change the domain of the action to return only scraps I created.

But the function you indicate only give me the values of the action without the domain, and even if the domain was returned but the function, it will not help me to change the domain and returns the action customized with my specific domain.

Maybe I misunderstanding something?

rousseldenis replied:
@jcoux The _for_xml_id is the core helper method to get the action. Moreover, it has sudo() in it too as some basic users don't have access to actions. So, please use it

ivantodorovich replied:
You'd use it like this:

```py
    action = self.env["ir.actions.actions"]._for_xml_id("stock.action_stock_scrap")
    action["domain"] = [("id", "in", scrap_ids)]
    return action
```

rousseldenis reviewed (changes requested):
LGTM. Just one change @jcoux

ivantodorovich reviewed (approved):
It seems that there's only 1 comment missing.
Pre-approving

rousseldenis commented:
@jcoux

jcoux commented:
Thanks for reminder and explanation: I will fix it today! :)

jcoux commented:
@rousseldenis 

Fixed with this commit: https://github.com/OCA/stock-logistics-workflow/commit/57490253eede8c4a7f91824a029842a81b3f4820
Then rebased and squashed

rousseldenis commented:
/ocabot merge patch
